File: backend/forum/src/main/java/com/grigor/forum/model/Permission.java
package com.grigor.forum.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@Builder
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class Permission {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id")
    private Integer id;

    @Basic
    @Column(name = "edit")
    private boolean edit;

    @Basic
    @Column(name = "post")
    private boolean post;

    @Basic
    @Column(name = "delete_comment")
    private boolean delete;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonBackReference("permission-user")
    private User user;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

}

File: frontend/src/app/models/permission.ts
import { Room } from "./room";
import { User } from "./user";

export class Permission {
    id!: number;
    post: boolean;
    edit: boolean;
    delete: boolean;
    room : Room | null;
    user : User | null;

    constructor(post:boolean, edit:boolean, deleteComment:boolean, room : Room, user : User) {
        this.post = post;
        this.edit = edit;
        this.delete = deleteComment;
        this.room = room;
        this.user = user;
    }
}
